lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -67,25 +67,32 @@ defmodule Thrift.Generator.StructBinaryProtocol do
   At the moment it also generates an experimental serializer that may be faster.
   """
   def struct_deserializer(%{fields: fields}, name, file_group) do
-    field_matchers = Enum.map(fields, fn %Field{name: name} ->
-      {name, Macro.var(name, nil)}
-    end)
-
-    struct_matcher = {:%, [], [name, {:%{}, [], field_matchers}]}
-
-    field_serializers = Enum.map(fields, fn %Field{name: name, type: type, id: id} ->
-      var = Macro.var(name, nil)
-      quote do
-        case unquote(var) do
-          nil ->
-            <<>>
-          _ ->
-            unquote([
-              quote do <<unquote(type_id(type, file_group)), unquote(id) :: size(16)>> end,
-              value_serializer(type, var, file_group)
-            ] |> Utils.merge_binaries |> Utils.simplify_iolist)
+    fields = Enum.reject(fields, &(&1.type == :void))
+
+    struct_matcher = case fields do
+                       [] ->
```

Comment:
That's me vs. preston. 
I'm using emacs's formatter and am subject to its whims. 

I profess a slight preference to to your suggestion, but emacs makes me unable to do that without doing it entirely by hand. I'm not going to do that because it would almost certainly result in 20 more iterations. 

At this point, I feel that many of the comments here are actually aimed at emacs's code style, and I'm considering moving off. 

To that end, if enforcing a specific style is important, (I don't think the core team agrees, btw) , then we should invest in an automatic formatting solution.